### PR TITLE
ml1

### PR DESCRIPTION
**Pull request by Meet1152**
Please Check the RF and NB

### CODE REVIEW

File: src/classification/machine learning.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+from sklearn.metrics import confusion_matrix
+from sklearn.metrics import accuracy_score
+
+class MachineLearning():
+
+    def __init__(self):
+
+        print("Loading dataset ...")
+
+        self.flow_dataset = pd.read_csv('/content/drive/MyDrive/COMS6200/NF-UNSW-NB15.csv')
```
Comment (dheerajram13) [inline src/classification/machine learning.py]: Change it to `pd.read_csv('../../data/NF-UNSW-NB15.csv')` because we'll be adding the CSV folder inside it from the data folder.

File: src/classification/naive_bayes.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+from sklearn.naive_bayes import GaussianNB
+from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, recall_score, f1_score
+
+class MachineLearning():
+
+    def __init__(self):
+        
+        print("Loading dataset ...")
+        
+        self.flow_dataset = pd.read_csv('/content/drive/MyDrive/COMS6200/NF-UNSW-NB15.csv')
```
Comment (dheerajram13) [inline src/classification/naive_bayes.py]: same here as well.

File: src/classification/naive_bayes.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    start = datetime.now()
+    
+    ml = MachineLearning()
+    ml.flow_training()
+
+    end = datetime.now()
+    print("Training time: ", (end-start)) 
+
+if __name__ == "__main__":
+    main()
```
Comment (dheerajram13) [inline src/classification/naive_bayes.py]: Add an empty a line at the end of the file.

File: src/classification/random_forests.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                    best_f1 = f1
+                    best_params = (n_estimators, criterion, max_depth)
+    
+    end = datetime.now()
+    print("Training time: ", (end-start)) 
+    print("Best Parameter Combination (n_estimators, criterion, max_depth):", best_params)
+    print("Best F1 Score:", best_f1)
+
+if __name__ == "__main__":
+    main()
```
Comment (dheerajram13) [inline src/classification/random_forests.py]: same here as well.

File: src/classification/machine learning.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
+from datetime import datetime
+from matplotlib import pyplot as plt
```
Comment (dheerajram13) [inline src/classification/machine learning.py]: Change the file name to machine_learning.py or ml.py  for better readability.

File: src/classification/machine learning.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    start = datetime.now()
+    ml.NB()
+    end = datetime.now()
+    print("LEARNING and PREDICTING Time: ", (end-start))
+
+    end_script = datetime.now()
+    print("Script Time: ", (end_script-start_script))
+
+if __name__ == "__main__":
+    main()
```
Comment (dheerajram13) [inline src/classification/machine learning.py]: Add an empty line at the end of the file.

File: src/classification/random_forests.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+    def flow_training(self, n_estimators=10, criterion='entropy', max_depth=None):
+        print("Flow Training with n_estimators={}, criterion={}, max_depth={} ...".format(n_estimators, criterion, max_depth))
+        
+        X_flow = self.flow_dataset.iloc[:, :-1].values
+        X_flow = X_flow.astype('float64')
+
+        y_flow = self.flow_dataset.iloc[:, -1].values
+
+        X_flow_train, X_flow_test, y_flow_train, y_flow_test = train_test_split(X_flow, y_flow, test_size=0.25, random_state=0)
```
Comment (dheerajram13) [inline src/classification/random_forests.py]: Can you please try it with a different test size like 0.3?

**Review by dheerajram13 — CHANGES_REQUESTED** [review]
Please complete the requested changes and try to implement the SVM with linear SVC class  

```
from sklearn.svm import LinearSVC 

cls = LinearSVC(random_state=0, tol=1e-5)
model = cls.fit(X_train, y_train.ravel())
```

File: src/classification/random_forests.py (hunk 7 above)
Comment (Meet1152) [reply src/classification/random_forests.py]: Will do it after checking the accuracy with it.

**Comment by Meet1152** [comment]
> Please complete the requested changes and try to implement the SVM with linear SVC class
> 
> ```
> from sklearn.svm import LinearSVC 
> 
> cls = LinearSVC(random_state=0, tol=1e-5)
> model = cls.fit(X_train, y_train.ravel())
> ```

File: src/classification/naive_bayes.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
+from datetime import datetime
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn.naive_bayes import GaussianNB
+from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, recall_score, f1_score
+
+class NBClassifier:
+
+    def __init__(self, dataset_path, dataset_size=400000):
```
Comment (dheerajram13) [inline src/classification/naive_bayes.py]: Use a constant for the _400000_ value instead of direct assignment and add a comment why we took that so that we could refer quickly in the future.

**Review by dheerajram13 — CHANGES_REQUESTED** [review]
LGTM! Please make that change and push it.
